src/contract_wrappers/modules/checkpoint/voting/advanced_p_l_c_r_voting_checkpoint_wrapper/common.ts

```diff
@@ -908,23 +935,30 @@ export default abstract class AdvancedPLCRVotingCheckpointCommon extends ModuleC
       return e.args;
     });
 
-    const onlyBallot = _.filter(args, e => {
+    const onlyBallot = _.find(args, e => {
       /* eslint-disable no-underscore-dangle */
       return e._ballotId.toNumber() === params.ballotId;
     });
+    assert.assert(onlyBallot !== undefined, ErrorCode.NotFound, 'Ballot id not found');
 
     const keccakKeys = ['uint256'];
-    params.choices.forEach(() => {
+    const bgVotes = params.choices.map(e => {
       keccakKeys.push('uint256');
+      return valueToWei(new BigNumber(e), FULL_DECIMALS).toString();
     });
-    const hash = ethersUtils.solidityKeccak256(keccakKeys, params.choices.concat(params.salt));
-    assert.assert(onlyBallot[0]._secretHash === hash, ErrorCode.InvalidData, 'Invalid vote');
+    const hash = ethersUtils.solidityKeccak256(keccakKeys, [...bgVotes, params.salt]);
+
+    assert.assert(
+      (onlyBallot as AdvancedPLCRVotingCheckpointVoteCommitEventArgs_3_1_0)._secretHash === hash,
```

Comment:
If the purpose of this type assertion is to make sure that `onlyBallot` isn't undefined, you can use a different (more succint) notation:

`assert.assert(onlyBallot!._secretHash === hash, /* rest of the params */);`

(Note the exclamation point after `onlyBallot`. That tells the compiler that you are sure that the value will be defined at that point)